lib/rack/proxy.rb

```diff
@@ -42,8 +42,9 @@ def reconstruct_header_name(name)
     def initialize(opts = {})
       @streaming = opts.fetch(:streaming, true)
       @ssl_verify_none = opts.fetch(:ssl_verify_none, false)
-      @backend = URI(opts[:backend]) if opts[:backend]
+      @backend = opts[:backend] ? URI(opts[:backend]) : nil
```

Comment:
I made the change because while running the tests each of those two lines produced warnings.  By using this style the warnings in the tests went away.  Not a problem if you prefer the previous style.